phpari.php

```diff
@@ -75,18 +75,53 @@ class phpari
 
 	/**
 	 * @param null $stasisApplication
-	 * @param string $configFile
+	 * @param <mixed> $config
 	 *
 	 * Returns an array containing 5 objects: WebSocket, Pest, EventLoopFactory, Logger, StasisEventHandler
 	 *
 	 */
-	public function __construct($stasisApplication = NULL, $configFile = "phpari.ini")
+	public function __construct($stasisApplication = NULL, $config = "phpari.ini")
 	{
 		try {
-
-			/* Get our configuration */
-			$this->configuration = (object)parse_ini_file($configFile, TRUE);
-
+			/* if config is an array pull configuration from the array */
+			if(is_array($config)) {
+				$asteriskAriConfig = array_merge(
+					array(
+						'username' => 'asterisk',
+						'password' => 'asterisk',
+						'host'     => '127.0.0.1',
+						'port'     => '8088',
+						'endpoint' => '/ari',
+						'transport' => 'ws',
+					),
+					isset($config['asterisk_ari']) ? $config['asterisk_ari'] : array()
+				);
+				$generalConfig = array_merge(
```

Comment:
You are mixing array styles from multiple PHP versions. PHPARI was designed to work with a minimum PHP version of 5.3, the `array_merge` styling you are mixing is version 5.4 in line 100 and then 5.3 in line 89. As the library is designed to work with machines running FreePBX and other distros, this code will most probably break - as most of them are still at PHP 5.3 (at least, not the extremely modern ones).

Also, code style inconsistencies can become a code readability issue along the way.